packages/backend/routes/devcon.js

```diff
@@ -0,0 +1,94 @@
+const express = require("express");
+const { ethers } = require("ethers");
+const db = require("../services/db/db");
+const { verifySignature } = require("../utils/sign");
+
+const router = express.Router();
+const SRE_BACKEND = "https://scaffold-directory-dev.ew.r.appspot.com";
+
+/**
+ * Check if a builder is eligible for a Devcon voucher based on their status and creation date
+ * Eligibility criteria:
+ * 1. BG batch members created before end of August 2024 (type: batch)
+ * 2. BG member created before end of August 2024 (type: builder)
+ * 3. SRE builders with at least 3 challenges completed and joined before end of August 2024 (type: builder)
+ */
+async function checkEligibility(builderAddress) {
+  const END_OF_AUGUST_2024 = 1725148799000;
+  const builder = await db.findUserByAddress(builderAddress);
+
+  if (builder?.data?.builderBatch && builder?.data?.creationTimestamp < END_OF_AUGUST_2024) {
+    return { isEligible: true, type: "batch" };
+  }
+
+  if (builder.exists && builder?.data?.creationTimestamp < END_OF_AUGUST_2024) {
+    return { isEligible: true, type: "builder" };
+  } else {
+    const response = await fetch(`${SRE_BACKEND}/builders/${builderAddress}`);
+
+    if (response.status === 404) {
+      return { isEligible: false };
+    }
+
+    if (response.status === 200) {
+      const data = await response.json();
+      const numberOfChallenges = Object.keys(data.challenges).length;
+      if (numberOfChallenges >= 3 && data.creationTimestamp < END_OF_AUGUST_2024) {
+        return { isEligible: true, type: "builder" };
+      }
```

Comment:
Probably wrap this inside try/catch, because when I tried making a request with an address which is not present in DB the server crashed: 

![Screenshot 2024-09-02 at 2 41 44 PM](https://github.com/user-attachments/assets/8a4941e0-1189-4f66-ba26-e9a055fd1947)


Also maybe not related to this but curious how this scenario is handled in production? Like if some unhandled error occurs in prod do we have to manually restart the backend?

Reply:
That error is because we were missing a check: fixed with `Object.values(data?.challenges || {})`

But in any case, we should wrap fetch request in try/catch. Just did in

https://github.com/scaffold-eth/buidlguidl-v3/pull/113/commits/b03bfdb9371ea70a22541c79803c39678c7393b2

Thanks!

> Also maybe not related to this but curious how this scenario is handled in production? Like if some unhandled error occurs in prod do we have to manually restart the backend?

Yes, this is how it works haha. e.g. if you use `pm2` it will restart the server when is crashes. In this case we are using GCP APP Engines, and it handles that for you.

The way to go is to check the logs and see when this happens... and fix!

Reply:
> Yes, this is how it works haha. e.g. if you use pm2 it will restart the server when is crashes. In this case we are using GCP APP Engines, and it handles that for you.

> The way to go is to check the logs and see when this happens... and fix!

Ohhhhh makes sense! Thanks for the explanation!